File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/KillJobRequest.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.constant.StringConstants
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class KillJobRequest(sessionId: String = StringConstants.EMPTY) {
}

object KillJobRequest {
  implicit def serialize(src: KillJobRequest): Array[Byte] = {
    Deepspeed.KillJobRequest.newBuilder()
      .setSessionId(src.sessionId)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): KillJobRequest = {
    val proto = Deepspeed.KillJobRequest.parseFrom(byteString)
    KillJobRequest(sessionId = proto.getSessionId)
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/KillJobRequest.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class KillJobRequest implements RpcMessage {

    Logger log = LoggerFactory.getLogger(KillJobRequest.class);
    private String sessionId;

    @Override
    public byte[] serialize() {
        Deepspeed.KillJobRequest.Builder builder = Deepspeed.KillJobRequest.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.KillJobRequest killJobRequest = Deepspeed.KillJobRequest.parseFrom(data);
            this.sessionId = killJobRequest.getSessionId();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
